api/src/main/java/org/openmrs/module/reporting/common/SqlIterator.java

```diff
@@ -0,0 +1,90 @@
+package org.openmrs.module.reporting.common;
+
+import org.openmrs.module.reporting.dataset.DataSetColumn;
+import org.openmrs.module.reporting.dataset.DataSetRow;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class SqlIterator implements Iterator<DataSetRow> {
+    private ResultSet resultSet;
+    private List<DataSetColumn> columns;
+    private Statement statement;
+    private Connection connection;
+
+    public SqlIterator(ResultSetMetaData metadata, ResultSet resultSet, Connection connection,Statement statement) throws SQLException {
+        this.resultSet = resultSet;
+        this.connection = connection;
+        this.statement = statement;
+        this.createDataSetColumns(metadata);
+    }
+
+    public boolean hasNext() {
+        try {
+            if(resultSet.isLast()){
```

Comment:
The code looks better if you put spaces around the `if` and `else` statements.

Reply:
Done. 
But the OpenMRS projects shouldn't had a configuration to autoformat when building?

Reply:
Most modules do not have.